File: little-bee/controller/router/driver.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_DRIVER = "/config/driver"
const URL_ROUTER_DRIVER_BY_ID = "/config/driver/:driver-id"
const URL_ROUTER_DRIVER_QUERY = "/config/driver/query"
const URL_ROUTER_DRIVER_ALL = "/config/driver/all"

func DriverRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_DRIVER, config.CreateDriver)
	r.GET(URL_ROUTER_DRIVER_BY_ID, config.RetrieveDriver)
	r.PUT(URL_ROUTER_DRIVER, config.UpdateDriver)
	r.DELETE(URL_ROUTER_DRIVER_BY_ID, config.DeleteDriver)
	r.PUT(URL_ROUTER_DRIVER_BY_ID, config.DeleteDriverWithFlag)
	r.POST(URL_ROUTER_DRIVER_QUERY, config.QueryDriver)
	r.GET(URL_ROUTER_DRIVER_ALL, config.GetAllDrivers)
	return r
}

File: frontend/console/src/api/driver.js
import Axios from "@/http";

export function createDriver(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/driver', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveDriver(driverId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/driver/${ driverId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateDriver(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/driver', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteDriver(driverId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/driver/${ driverId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteDriverWithFlag(driverId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/driver/${ driverId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllDrivers() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/driver/all').then(res => {
            resolve(res)
        })
    })
}

export function queryDriver (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/driver/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}